internal/controllers/addon/addon_instance_reconciler.go

```diff
@@ -87,5 +102,23 @@ func (r *addonInstanceReconciler) reconcileAddonInstance(
 		currentAddonInstance.OwnerReferences = desiredAddonInstance.OwnerReferences
 		return r.client.Update(ctx, currentAddonInstance)
 	}
+
+	// If addon's operator choose to onboard addon instance into their workflow.
+	// Wait for the addon instance to be installed.
+	if addon.Spec.InstallAckRequired {
```

Comment:
Instead of checking this here, can we perform this check when we set the installed and available condition.

Reply:
Installed condition in addon instance != true is not really an error condition.

Reply:
I've moved the check to where installation is handled, please take a look